inputsource.cpp

```diff
@@ -218,7 +294,34 @@ void InputSource::openFile(const char *filename)
         sampleAdapter = std::unique_ptr<SampleAdapter>(new ComplexF32SampleAdapter());
     }
 
-    std::unique_ptr<QFile> file(new QFile(filename));
+    QString dataFilename;
+
+#if ENABLE_SIGMF
+    QString metaFilename;
+
+    if (suffix == "sigmf-meta") {
+        dataFilename = fileInfo.path() + "/" + fileInfo.completeBaseName() + ".sigmf-data";
+        metaFilename = filename;
+        readMetaData(metaFilename.toUtf8().constData());
+    }
+    else if (suffix == "sigmf-data") {
+        dataFilename = filename;
+        metaFilename = fileInfo.path() + "/" + fileInfo.completeBaseName() + ".sigmf-meta";
+        readMetaData(metaFilename.toUtf8().constData());
+    }
+    else if (suffix == "sigmf") {
+        throw std::runtime_error("SigMF archives are not supported. Consider extracting a recording.");
+    }
+#else
+    if (suffix == "sigmf-meta" || suffix == "sigmf-data" || suffix == "sigmf") {
+        throw std::runtime_error("Support for SigMF recordings is not enabled");
```

Comment:
@miek If libsigmf is not found it will now refuse to open these files. One could argue that it should still be possible to open a `.sigmf-data` file as it actually contains samples. Any opinion on that? I could for example print a warning on the console in that case.

Reply:
I think that's OK. If it were changed to allow opening just the `.sigmf-data`, there would be no way to pick the right datatype.